plugins/main/server/plugin.ts

```diff
@@ -37,36 +44,32 @@ import {
   jobSanitizeUploadedFilesTasksRun,
 } from './start';
 import { first } from 'rxjs/operators';
+import { WazuhCore } from '../common/types';
 
-declare module 'opensearch_dashboards/server' {
+declare module '../../../src/core/server' {
   interface RequestHandlerContext {
     wazuh: {
       logger: Logger;
       plugins: PluginSetup;
-      security: any;
+      security: DashboardSecurity;
       api: {
-        client: {
-          asInternalUser: {
-            authenticate: (apiHostID: string) => Promise<string>;
-            request: (
-              method: string,
-              path: string,
-              data: any,
-              options: { apiHostID: string; forceRefresh?: boolean },
-            ) => Promise<any>;
-          };
-          asCurrentUser: {
-            authenticate: (apiHostID: string) => Promise<string>;
-            request: (
-              method: string,
-              path: string,
-              data: any,
-              options: { apiHostID: string; forceRefresh?: boolean },
-            ) => Promise<any>;
-          };
+        client: ApiClient;
+      };
+      server: {
+        info: {
+          name: string;
+          hostname: string;
+          port: number;
+          protocol: string;
         };
       };
     };
+    wazuhEndpointParams: {
```

Comment:
**nitpick**: these can not be present in some requests because these are added in a route decorator.